modules/core/src/adapter/attribute-utils/get-attribute-from-layouts.ts

```diff
@@ -110,49 +120,92 @@ function getAttributeFromShaderLayout(
 }
 
 function getAttributeFromBufferLayout(
-  bufferLayout: BufferLayout[],
+  bufferLayouts: BufferLayout[],
   name: string
-): Partial<AttributeInfo> | null {
-  for (const bufferMapping of bufferLayout) {
-    // Search interleaved attributes for buffer mapping. 
-    // The name of the buffer might be the same as one of the interleaved attributes.
-    let nextByteOffset = bufferMapping.byteOffset || 0;
-    let byteStride = 0;
-    for (const interleavedMapping of bufferMapping.attributes || []) {
-      const info = decodeVertexFormat(interleavedMapping.format);
-      byteStride += info.byteLength;
+): BufferAttributeInfo | null {
+  // Check that bufferLayouts are valid (each either has format or attribute)
+  checkBufferLayouts(bufferLayouts);
+
+  let bufferLayoutInfo = getAttributeFromShortHand(bufferLayouts, name);
+  if (bufferLayoutInfo) {
+    return bufferLayoutInfo;
+  }
+
+  bufferLayoutInfo = getAttributeFromAttributesList(bufferLayouts, name);
+  if (bufferLayoutInfo) {
+    return bufferLayoutInfo;
+  }
+
+  // Didn't find...
+  log.warn(`layout for attribute "${name}" not present in buffer layout`);
+  return null;
+}
+
+/** Check that bufferLayouts are valid (each either has format or attribute) */
+function checkBufferLayouts(bufferLayouts: BufferLayout[]) {
+  for (const bufferLayout of bufferLayouts) {
+    if (
+      (bufferLayout.attributes && bufferLayout.format) ||
+      (!bufferLayout.attributes && !bufferLayout.format)
+    ) {
+      log.warn(`BufferLayout ${name} must have either 'attributes' or 'format' field`);
     }
-    for (const interleavedMapping of bufferMapping.attributes || []) {
-      const byteOffset = nextByteOffset;
-      nextByteOffset +=
-        interleavedMapping?.byteOffset ||
-        decodeVertexFormat(interleavedMapping.format).byteLength;
-      // const interleavedMapping = bufferMapping.attributes?.find(attr => attr.name === name);
-      if (interleavedMapping.name === name) {
-        return {
-          name: bufferMapping.name,
-          bufferName: name,
-          vertexFormat: interleavedMapping.format,
-          byteOffset: interleavedMapping.byteOffset ?? byteOffset,
-          byteStride: bufferMapping.byteStride || byteStride
-        };
-      }
+  }
+}
+
+/** Get attribute from format shorthand if specified */
+function getAttributeFromShortHand(
+  bufferLayouts: BufferLayout[],
+  name: string
+): BufferAttributeInfo | null {
+  for (const bufferLayout of bufferLayouts) {
+    if (bufferLayout.format && bufferLayout.name === name) {
+      return {
+        attributeName: bufferLayout.name,
+        bufferName: name,
+        stepMode: bufferLayout.stepMode,
+        vertexFormat: bufferLayout.format,
+        // If offset is needed, use `attributes` field.
+        byteOffset: 0,
```

Comment:
Thinking of the user-supplied binary buffer case in deck.gl - does this mean we have to format bufferLayout differently depending on whether offset is 0?